packages/react/src/auto/polaris/inputs/PolarisAutoInput.tsx

```diff
@@ -0,0 +1,107 @@
+import type { TextFieldProps } from "@shopify/polaris";
+import { Checkbox, TextField } from "@shopify/polaris";
+import React from "react";
+import { useController } from "react-hook-form";
+import type { GadgetEnumConfig, GadgetFieldType } from "../../../internal/gql/graphql.js";
+import { FieldType } from "../../../metadata.js";
+import { useFormFields } from "../../AutoForm.js";
+import { useAutoFormMetadata } from "../../AutoFormContext.js";
+import { PolarisDateTimePicker } from "../PolarisDateTimePicker.js";
+import { PolarisFileInput } from "../PolarisFileInput.js";
+import { PolarisFixedOptionsCombobox } from "../PolarisFixedOptionsCombobox.js";
+import { PolarisJSONInput } from "../PolarisJSONInput.js";
+import { PolarisRolesCombobox } from "../PolarisRolesCombobox.js";
+import { PolarisBelongsToInput } from "./PolarisBelongsToInput.js";
+
+const FieldTypeToInputType: Partial<Record<GadgetFieldType, TextFieldProps["type"]>> = {
+  [FieldType.Number]: "number",
+  [FieldType.Email]: "email",
+  [FieldType.Password]: "password",
+  [FieldType.EncryptedString]: "password",
+};
+
+export const PolarisAutoInput = (props: { field: string }) => {
```

Comment:
What's the difference between this and PolarisFormInput

Reply:
FormInput takes the metadata as a prop, and AutoInput gets the metadata from the context. I'd like to eventually completely replace FormInput with AutoInput, but would prefer to do that in a followup PR